api/serverless.yml

```diff
@@ -50,7 +49,7 @@ resources:
       Type: AWS::IAM::Role
       Properties:
         Path: /my/default/path/
-        RoleName: DefaultNameGuardRole # required if you want to use 'serverless deploy --function' later on
+        RoleName: DefaultNameGuardRole-${self:custom.stage}
```

Comment:
Instead of having one role (and complicating deployment for more than one env), we have role and function per env